git-config/src/file/git_config.rs

```diff
@@ -147,18 +148,57 @@ impl<'event> GitConfig<'event> {
     #[inline]
     pub fn from_paths(paths: &[&Path]) -> Result<Self, ParserOrIoError<'static>> {
         let mut config = Self::new();
-
-        for path in paths {
-            let other = Self::open(path)?;
-            for (section_id, section_header) in other.section_headers {
-                config.push_section(
-                    section_header.name.0.to_owned(),
-                    section_header.subsection_name.to_owned(),
-                    other.sections[&section_id].clone(),
-                );
+        let mut seen = HashSet::new();
+        let paths: Vec<PathBuf> = paths
+            .iter()
+            .map(|p| PathBuf::from(p.to_str().expect("invalid utf-8 path")))
+            .collect();
+        const MAX_DEPTH: u8 = 10;
+
+        fn from_paths_req(
+            paths: Vec<PathBuf>,
+            seen: &mut HashSet<PathBuf>,
+            config: &mut GitConfig,
+            depth: u8,
+        ) -> Result<(), ParserOrIoError<'static>> {
+            if depth >= MAX_DEPTH {
+                return Ok(());
+            }
+            for path in paths.iter() {
+                let other = GitConfig::open(path)?;
+                let mut section_ids: Vec<_> = other.section_headers.keys().collect();
+                section_ids.sort();
+                for section_id in section_ids {
+                    let section_header = &other.section_headers[&section_id];
+                    if section_header.name.0 == "include" {
```

Comment:
could not use `multi_value` since it breaks section ordering

Reply:
That's good to know, thanks for the note.

Maybe we keep that in mind and ask ourselves if maintaining section ordering is something we would want or need, especially if this affects the [last-one wins](https://github.com/git/git/blob/main/config.c#L2267:L2276) semantics.